src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4306,6 +4308,9 @@ def set_vmss_extension(cmd, resource_group_name, vmss_name, extension_name, publ
     if not extension_instance_name:
         extension_instance_name = extension_name
 
+    if extension_name == 'CodeIntegrityAgent' and enable_auto_upgrade is None:
+        enable_auto_upgrade = True
```

Comment:
Could you please help confirm with the server team that `GuestAttestation` needs to be supported for automatic updates in this case?

Reply:
Confirmed and not sure if `GuestAttestation` needs to be supported in this case.
https://github.com/Azure/azure-cli/issues/27244#issuecomment-1719721148